Interpreter.c

```diff
@@ -76,8 +76,71 @@ void Interpreter_pushInteger(short integerValue) {
 	Interpreter_success_(YES);
 }
 
+/** 
+ * Page 671
+ * convert back and forth between SmallInteger and LargePositiveInteger
+ */
+ObjectPointer Interpreter_positive16BitIntegerFor(short integerValue) {
+	ObjectPointer result = NilPointer;
+	
+	/* if its negative, we fail */
+	if( integerValue < 0 ) {
+		Interpreter_primitiveFail();
+		return NilPointer; /* caller shouldn't rely on this */
+	}
+	
+	/* if its small enough we can just return get an Integer object version */
+	if( ObjectMemory_isIntegerValue(integerValue) ) {
+		return ObjectMemory_integerObjectOf(integerValue);
+	}
+	
+	/* Its too big so we create a LargePositiveInteger */
+	result = ObjectMemory_instantiateClass_withBytes(ClassLargePositiveIntegerPointer, 2);
+	ObjectMemory_storeByte_ofObject_withValue(0, result, Interpreter_lowByteOf(integerValue));
+	ObjectMemory_storeByte_ofObject_withValue(1, result, Interpreter_highByteOf(integerValue));
+	return result;
+}
+
+short Interpreter_positive16BitValueOf(ObjectPointer integerPointer) {
+	short result = 0;
+	/* if its a small integer, just return integerValue */
+	if( ObjectMemory_isIntegerObject(integerPointer) ) {
+		result = ObjectMemory_integerValueOf(integerPointer);
+		/* its negative fail, duplicated below */
+		if( result < 0 ) {
+			Interpreter_primitiveFail();
+			return 0; /* caller shouldn't rely on this */
```

Comment:
yup, fair to return any old value after the routine has failed. The interpreter should check the success register rather than checking the result makes sense.